includes/abstracts/llms-abstract-generator-posts.php

```diff
@@ -649,7 +679,7 @@ protected function set_featured_image( $url_or_raw, $post_id ) {
 
 		$image_url = ( is_array( $url_or_raw ) && ! empty( $url_or_raw['featured_image'] ) ) ? $url_or_raw['featured_image'] : $url_or_raw;
 
-		if ( $image_url ) {
+		if ( is_string( $image_url ) ) {
```

Comment:
```suggestion
		if ( is_string( $image_url ) && $image_url ) {
```
right? or we're just assuming that variable will never be empty?

Reply:
I think it could actually be an empty string, which we wouldn't want to pass in so you're right, but I'm going to reverse the checks so it's `$image_url && is_string(...)`